### PR TITLE
Allow reusing projects between runs of logical replication benchmarks

### PR DESCRIPTION
**Pull request by save-buffer**
## Problem
We want to exercise pageservers with LR over the longterm. Currently we do not allow reusing projects between runs of these tests. (https://github.com/neondatabase/neon/issues/8317)

## Summary of changes
This PR lets you specify an env var to reuse projects using a fixture. This also cascades into a small refactor of the tests to hide some of the raw calls to the API.

### CODE REVIEW

File: test_runner/fixtures/neon_fixtures.py [hunk 1]
```diff
@@ -3158,6 +3158,67 @@ def __exit__(
         pass
 
 
+class RemoteNeonApiEndpoint:
```
Comment (tristan957) [inline test_runner/fixtures/neon_fixtures.py:3161]: I would put this class definition in the noen_api.py file.

Comment (tristan957) [reply test_runner/fixtures/neon_fixtures.py:3161]: and maybe just call it RemoteEndpoint

File: test_runner/fixtures/neon_fixtures.py [hunk 2]
```diff
@@ -3158,6 +3158,67 @@ def __exit__(
...
+            self.project_id = project["project"]["id"]
+            self.endpoint_id = project["endpoints"][0]["id"]
+            self.connstr = project["connection_uris"][0]["connection_uri"]
+            self.pgbench_env = connection_parameters_to_env(
+                project["connection_uris"][0]["connection_parameters"]
+            )
+            self.is_new = True
+        else:
+            project = neon_api.get_project_details(project_id)
+            if int(project["project"]["pg_version"]) != int(pg_version):
```
Comment (tristan957) [inline test_runner/fixtures/neon_fixtures.py:3176]: Comparing ints probably doesn't matter here

File: test_runner/fixtures/neon_fixtures.py [hunk 3]
```diff
@@ -3158,6 +3158,67 @@ def __exit__(
...
+        self.neon_api.restart_endpoint(self.project_id, self.endpoint_id)
+        self.neon_api.wait_for_operation_to_finish(self.project_id)
+
+    def get_synthetic_storage_size(self) -> int:
+        return int(
+            self.neon_api.get_project_details(self.project_id)["project"]["synthetic_storage_size"]
+        )
+
+
+@pytest.fixture(scope="function")
```
Comment (tristan957) [inline test_runner/fixtures/neon_fixtures.py:3205]: Should the scopes be session?

File: test_runner/fixtures/neon_fixtures.py (hunk 2 above)
Comment (save-buffer) [reply test_runner/fixtures/neon_fixtures.py:3176]: One is a string and one is a `pgversion`, so they have to get casted to the same type (probably doesn't matter exactly what that type is).

File: test_runner/fixtures/neon_fixtures.py (hunk 3 above)
Comment (save-buffer) [reply test_runner/fixtures/neon_fixtures.py:3205]: Yep probably can be, changed.

File: test_runner/fixtures/neon_fixtures.py (hunk 1 above)
Comment (save-buffer) [reply test_runner/fixtures/neon_fixtures.py:3161]: We already have a `RemotePostgres`, which may or may not be Neon, so I wanted the name to specify that it was based on `NeonApi`. I'll shorten it to `NeonApiEndpoint`

File: test_runner/fixtures/neon_fixtures.py (hunk 3 above)
Comment (save-buffer) [reply test_runner/fixtures/neon_fixtures.py:3205]: Nevermind, it can't be because it relies on pgversion, which is function scoped

**Review by tristan957 — APPROVED** [review]
Good job!

**Pull request merged**